### PR TITLE
connectors-ci: deprecate slash test

### LINKED ISSUE #23978: connectors-ci: discard the use of `/test`
## Tell us about the problem you're trying to solve
The `connectors-ci test-connectors` command running in the CI is made to replace the use of the `/test` slash command from PR comments.
Once we're confident about the stability of this workflow we should remove the `/test` command and remove documentation about its usage.

This future legacy workflow is declared here:
https://github.com/airbytehq/airbyte/blob/master/.github/workflows/test-command.yml#L1

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes #23978

This PR retires the /test command by:
* Disabling the original `/test`. When using `/test` users will get a deprecation message.
* Replacing `/test` with `/legacy-test` for use cases not yet ported to `airbyte-ci` like testing connectors against CDK or CAT change.
* Disabling the "legacy" nightly builds workflow and the slack reporting workflow that come along
* Enable the `PRODUCTION` flag on new automated test workflow, this will enable bot comments on PRs and remove the "please ignore" prefix on automated CI checks
* Remove the connector metadata validation workflow: metadata validation now happens in the automated tests
* Update our documentation

## User impact
Merging this PR will lead to immediate retirement of the /test command and the switch over to automated testing.
As we can't progamatically define "required CI checks to merge" we'll manually make the "Connectors CI tests" check required in Github settings page.

**For smooth transition to the new nightly build it's preferrable to merge this PR after 3AM UTC.**
**Note: if anything goes wrong reverting this PR should be the single thing to do to rollback.**

### CODE REVIEW

File: .github/workflows/legacy-test-command.yml
Comment (evantahler) [inline .github/workflows/legacy-test-command.yml:1]: :thumbsup: we can still use the old /test command if needed

File: .github/workflows/test-command.yml
Comment (evantahler) [inline .github/workflows/test-command.yml:30]: 👍 users of /test get a warning

**Review by evantahler — APPROVED** [review]
👍 from me! Good doc updates.
We are waiting until some pipeline+dagger updates to merge this

File: .github/workflows/connectors_weekly_build.yml
Comment (bnchrch) [inline .github/workflows/connectors_weekly_build.yml:68]: @alafanechere added the job key, depends on this merging into master https://github.com/airbytehq/airbyte/pull/27210

**Pull request merged**
